docs/record-encryption-quickstart/index.adoc

```diff
@@ -0,0 +1,254 @@
+----
+
+=== Install the Strimzi Operator
+
+We are going to use Strimzi to deploy the Kafka Cluster that will be proxied.
+Let's install Strimzi next:
+
+[source,shell]
+----
+helm repo add --force-update strimzi https://strimzi.io/charts/
+helm upgrade --install strimzi-operator strimzi/strimzi-kafka-operator --namespace kafka --create-namespace --version 0.47.0
+----
+
+=== Install the Kroxylicious Operator
+
+First, create a temporary directory for some working files.  We use a directory called `ko-install` beneath `/tmp`, but you can use any name and location you like.
+
+[source,shell]
+----
+mkdir /tmp/ko-install
+cd /tmp/ko-install
+----
+
+Now let's install the Kroxylicious Operator
+
+[source,shell]
+----
+curl --location https://github.com/kroxylicious/kroxylicious/releases/download/v0.13.0/kroxylicious-operator-0.13.0.zip --output kroxylicious-operator.zip
+unzip kroxylicious-operator.zip
+kubectl apply -f install
+kubectl wait --namespace kroxylicious-operator deployment/kroxylicious-operator --for=condition=Available=True --timeout=300s
+----
+
+== Deploy the Kafka Cluster
+
+Now we need a Kafka Cluster.
+This is the Kafka Cluster that will be proxied.
+We'll just use one from the Strimzi Quickstart.
+It will create a cluster in the `kafka` namespace
+
+[source,shell]
+----
+kubectl apply -n kafka -f https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/refs/tags/0.47.0/examples/kafka/kafka-single-node.yaml
+kubectl wait -n kafka kafka/my-cluster --for=condition=Ready --timeout=300s
+----
+
+We'll need the boostrap server address of the cluster later, so let's assign a variable containing it now.
+
+[source,shell]
+----
+DIRECT_CLUSTER_BOOTSTRAP_SERVERS=$(kubectl get -n kafka kafka my-cluster -o=jsonpath='{.status.listeners[0].bootstrapServers}')
+----
+
+
+== Deploy the Proxy with Record Encryption
+
+Next, we'll deploy a Kroxylicious proxy instance using the record encryption example included in the install zip you download above.
+
+It will create an instance of Kroxylicious that will proxy the kafka cluster you created a above, and configure it to
+apply the Record Encryption filter to encrypt as they are sent by producers and decrypt them as they get fetched by consumers.
+The proxy will be created the `my-proxy` namespace
+
+[source,shell]
+----
+kubectl apply -f examples/record-encryption/
+----
+
+You need to update the example to know about your chosen KMS.
+
+If you are using Vault:
+
+[source,shell]
+----
+kubectl patch -n my-proxy kafkaprotocolfilters.kroxylicious.io encryption --patch '{"spec":{"configTemplate":{"kms":"VaultKmsService","kmsConfig":{"vaultToken":{"password":"myroottoken"},"vaultTransitEngineUrl":"http://vault.vault.svc.cluster.local:8200/v1/transit"}, "selector" : "TemplateKekSelector", "selectorConfig" : {"template": "KEK_$(topicName)"}}}}' --type merge
```

Comment:
I'm planning to change the example to bake the TemplateKekSelector into the config.  We've only got one right now, so we may as well hardcoded in the example.